src/Control/Distributed/Process/Tests/CH.hs

```diff
@@ -1316,6 +1318,111 @@ testUnsafeSendChan TestTransport{..} = do
 
   takeMVar clientDone
 
+testRegistryMonitoring :: TestTransport -> Assertion
+testRegistryMonitoring TestTransport{..} = do
+  localNode <- newLocalNode testTransport initRemoteTable
+  remoteNode <- newLocalNode testTransport initRemoteTable
+  return ()
+
+  -- Local process. Test if local process will be removed from
+  -- registry when it dies.
+  box <- newEmptyMVar
+  runProcess localNode $ do
+    pid <- spawnLocal $ do
+                expect
+    register "test" pid
+    tpid <- whereis "test"
+    if tpid == Just pid
+      then do _ <- monitor pid
+              send pid ()
+              ProcessMonitorNotification{} <- expect
+              tpid1 <- whereis "test"
+              liftIO $ putMVar box (Nothing == tpid1)
+      else liftIO $ putMVar box False
+  True <- takeMVar box     
+  return ()
+
+  -- Remote process. Test if remote process entry is removed
+  -- from registry when process dies.
+  remote1 <- testRemote remoteNode 
+  runProcess localNode $
+    let waitpoll = do
```

Comment:
I'm not sure I know how to write this test then. Because `linking` or monitoring will interfere with test, I.e. if I'll link to process then process is guaranteed to be removed from registry, and I want to test if it will be removed even if no one is linked to it.
I'll need to think about this